bot.py

```diff
@@ -56,19 +57,215 @@
+    embed.add_field(name='\u200b', value=totalString, inline=True)
+    ran = random.randint(1, len(allImages))
+    embed.set_thumbnail(url=str(allImages[ran - 1]))
+    await message.channel.send(embed=embed)
+    embed.clear_fields()
+
+
+async def civDraft(self, message, pl=0, cipp=0):
+    global civ_dict
+
+    def is_valid_number(m):
+        return m.author == message.author and m.content.isdigit()
+
+    def is_number(f):
+        try:
+            int(f)
+            return True
+        except:
+            return False
+
+    # Ask for number of players if not provided
+    if pl == 0:
+        await message.channel.send('How many players are playing?')
+        try:
+            number_players = await self.wait_for('message', check=is_valid_number)
+        except asyncio.TimeoutError:
+            return
+    else:
+        number_players = pl
+
+    # Ask for number of civs per player if not provided
+    if cipp == 0:
+        await message.channel.send('How many civilizations do you want to draft?')
+        try:
+            number_civs = await self.wait_for('message', check=is_valid_number)
+        except asyncio.TimeoutError:
+            return
+    else:
+        number_civs = cipp
+
+    if isinstance(number_civs, str) == False:
+        number_civs = number_civs.content
+    if isinstance(number_players, str) == False:
+        number_players = number_players.content
+
+    if is_number(number_players):
+        pass
+    else:
+        await message.channel.send(message.author.mention + ' Unknown what `' + number_players + '` means in context')
+        return
+    if is_number(number_civs):
+        pass
+    else:
+        await message.channel.send(message.author.mention + ' Unknown what `' + number_civs + '` means in context')
+        return
+
+    unusedCivs = []
+    for i in civ_dict:
+        unusedCivs.append(int(i['ID']))
+
+    randomIDArray = []
+    civsLen = len(unusedCivs)
+    for x in range(0, int(number_players)):
+        randomIDArray.insert(x, [])
+        for y in range(0, int(number_civs)):
+            exit = False
+            while exit == False:
+                randomInt = random.randint(1, civsLen)
+                try:
+                    unusedCivs.remove(randomInt)
+                    randomIDArray[x].append(randomInt)
+                    exit = True
+                except:
+                    pass
+
+    for p in range(0, len(randomIDArray)):
+        await displayEmbedCivs(message, p, randomIDArray[p])
+
+
+async def hoiDraft(self, message, pl=0, cpp=0, notRequireMajor=False):
+    global hoi_dict
+
+    def is_valid_number(m):
+        return m.author == message.author and m.content.isdigit()
+
+    def is_number(f):
+        try:
+            int(f)
+            return True
+        except:
+            return False
+
+    # Ask for number of players if not provided
+    if pl == 0:
+        await message.channel.send('How many players are playing?')
```

Comment:
The asking of "how many players" is done a couple times.  Actually just asking anything is done a few times.
Consider making a function to ask for a response where it can take a parameter of what you are asking, and take another parameter to validate that the response is if the right type.

Then if you ever change or enhance the way you ask for and receive responses you only have a single place to do it.